django/db/models/fields/related_descriptors.py

```diff
@@ -195,6 +195,13 @@ def get_prefetch_querysets(self, instances, querysets=None):
         else:
             query = {"%s__in" % self.field.related_query_name(): instances}
         queryset = queryset.filter(**query)
+        # Since there can be only one object prefetched for each instance
+        # order by is insignificant and slows down the query.
+        # queryset = queryset.order_by()
+        # queryset was cloned by filter() just above, no need to re-clone it.
+        queryset._in_place_empty_order_by()
+        # No need to handle the case where the query is_sliced (LIMIT)
+        # because no filter can be added to a sliced query.
```

Comment:
I'm pretty sure you can simply do [the following](https://github.com/django/django/blob/b07e2d57a000d98c73492e5242fed91d502a780a/django/db/models/sql/query.py#L2272C9-L2286) to avoid cloning and work around the possible limit issue.

```suggestion
        queryset = queryset.filter(**query)
        # There can be only one object prefetched for each instance so
        # clear ordering if the query allows it without side effects.
        queryset.query.clear_ordering(force=False)
```

Reply:
Yes that's what I wrote in the ticket one hour ago. I will take this solution.

Reply:
Note that you need `clear_default=True` as I did. Otherwise it will not work.

Reply:
No need to explicit pass it; `clear_ordering(clear_default)` [default value is already `True`](https://github.com/django/django/blob/b07e2d57a000d98c73492e5242fed91d502a780a/django/db/models/sql/query.py#L2272C9-L2286).

Reply:
Indeed, sorry about that, it is here